### PR TITLE
add docs about how to find the builder

### PR DESCRIPTION
**Pull request by deitch**
Signed-off-by: Avi Deitcher <avi@deitcher.net>

**- What I did**

Added docs about the three ways to find the kernel builder image

**- How I did it**

Updated `docs/kernels.md`

**- How to verify it**

Just read it; it is docs.

**- Description for the changelog**

Document how to find the builder for kernels.

### CODE REVIEW

File: docs/kernels.md [hunk 1]
```diff
@@ -45,22 +45,42 @@ Most kernel modules are autoloaded with `mdev` but if you need to `modprobe` a m
...
 This section describes how to build external (out-of-tree) kernel
-modules. It is assumed you have the source available to those modules,
-and require the correct kernel version headers and compile tools.
+modules. You need the following to build external modules. All of
+these are to be built for a specific version of the kernel. For
+the examples, we will assume 5.10.104; replace with your desired
+version.
+
+* source available to your modules - you need to get those on your own
+* kernel development headers - available in the `linuxkit/kernel` as `kernel-dev.tar`, e.g. `linuxkit/kernel:5.10.104`
```
Comment (rn) [inline docs/kernels.md:54]: nit: "In the linuxkit/kernel" image

File: docs/kernels.md [hunk 2]
```diff
@@ -45,22 +45,42 @@ Most kernel modules are autoloaded with `mdev` but if you need to `modprobe` a m
...
+* OS with sources and compiler - this **must** be the exact same version as that used to compile the kernel
+
+You have 3 places to determine the correct version of the OS image with sources and compiler:
+
+* `-builder` tag added to the kernel version, e.g. `linuxkit/kernel:5.10.104-builder`
+* labels on the kernel image, e.g. `docker inspect linuxkit/kernel:5.10.104 | jq -r '.[].Config.Labels["org.mobyproject.linuxkit.kernel.buildimage"]'`
+* `/kernel-builder` file in the kernel image
+
+Use any of the above the the `<builder>`.o
```
Comment (rn) [inline docs/kernels.md:63]: nit, double "the". Also not a complete sentence

File: docs/kernels.md [hunk 3]
```diff
@@ -45,22 +45,42 @@ Most kernel modules are autoloaded with `mdev` but if you need to `modprobe` a m
...
-There is a [example](../test/cases/020_kernel/011_kmod_4.9.x), but
+There is an [example](../test/cases/020_kernel/011_kmod_4.9.x), but
 basically one can use a multi-stage build to compile the kernel
 modules:
 
 ```
-FROM linuxkit/kernel:4.9.33 AS ksrc
-FROM linuxkit/alpine:<hash> AS build
+FROM linuxkit/kernel:5.10.104 AS ksrc
+FROM linuxkit/alpine:2be490394653b7967c250e86fd42cef88de428ba AS build
```
Comment (rn) [inline docs/kernels.md:81]: should we default to the `-builder` case. It seems more intuitive and requires one less step, ie does not require looking up the alpine hash? The alpine hash  should only be necessary of we retag or something

**Review by rn — COMMENTED** [review]
Thanks for adding/updating the docs. As mentioned in one of the comments, I think it might easier to write/understand if we just default to the `-builder` images and only use the alpine images as backup. They are more like a implementation detail

File: docs/kernels.md (hunk 1 above)
Comment (deitch) [reply docs/kernels.md:54]: Good catch. Updated.

File: docs/kernels.md (hunk 2 above)
Comment (deitch) [reply docs/kernels.md:63]: Yeah, strange that. Fixed.

File: docs/kernels.md (hunk 3 above)
Comment (deitch) [reply docs/kernels.md:81]: Yes, that not only is a good point, but is exactly what it is there for. The others are backups... which turned out to be a good thing. I had to track one down last week that was missing the `-builder` tag, and could not remember how to do it. This saved me (and prompted me to write this doc).

I will reword this.

Comment (deitch) [reply docs/kernels.md]: Done

**Comment by deitch** [comment]
Restructured it to reference the `-builder` as primary. Take another look @rn

**Review by rn — APPROVED** [review]

**Pull request merged**